src/position.cpp

```diff
@@ -1156,12 +1157,33 @@ bool Position::see_ge(Move m, Value threshold) const {
       return true;
 
   bool opponentToMove = true;
-  occupied = pieces() ^ from ^ to;
+  occupied = pieces() ^ to;
 
-  // Find all attackers to the destination square, with the moving piece removed,
-  // but possibly an X-ray attacker added behind it.
+  if (gating_type(m) == NO_GATE_TYPE)
+      occupied ^= from;
+
+  // Find all attackers to the destination square including X-ray attacks
+  // if the move was not a gating move.
   Bitboard attackers = attackers_to(to, occupied) & occupied;
 
+  if (gating_type(m) != NO_GATE_TYPE)
+  {
+      // At this point 'attackers' always contains the 'from' square
+      // because the 'from' square is set in 'occupied' and the call
+      // to attackers_to believed the moving piece was still present.
+      // If the gated piece does not attack the 'to' square then make
+      // a correction.
+      if (!(PseudoAttacks[gating_type(m)][from] & to))
+          attackers ^= from;
+
+      // Create a copy of the piece bitboards array and put the gated
+      // piece in place of the original.
+      memcpy(tmpPiecesBB, byTypeBB, int(PIECE_TYPE_NB) * sizeof(Bitboard));
```

Comment:
Don't you need to allocate the memory for `tmpPiecesBB` before doing the `memcpy`? (Sorry, I rarely use memcpy, so I am not expert).

Reply:
Declaring `tmpPiecesBB[PIECE_TYPE_NB]` on line 1142 reserves space for `tmpPiecesBB` on the stack which will cease to be available when the function exits.